### PR TITLE
Plugin change to support Salesforce Security Token. 

### PR DESCRIPTION
**Pull request by shiva-cirus**
JIRA - Plugin-420.  Implementation of the Security token for SFDC Plugin. As discussed with Vinisha there is a TODO added due to previous error in the branch.

### CODE REVIEW

File: docs/Salesforce-batchsource.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ Configuration
 
 **Password:** Salesforce password.
 
+**Security Token:** Salesforce security Token. If the password does not contain the security token the plugin 
+will append the token before authenticating with salesforce.
+
+
```
Comment (rmstar) [inline docs/Salesforce-batchsource.md]: nit: remove extra newline

File: docs/Salesforce-streamingsource.md [hunk 2]
```diff
@@ -25,6 +25,10 @@ Configuration
 
 **Password:** Salesforce password.
 
+**Security Token:** Salesforce security Token. If the password does not contain the security token the plugin 
+will append the token before authenticating with salesforce.
+
+
```
Comment (rmstar) [inline docs/Salesforce-streamingsource.md]: nit: remove extra newline

File: docs/SalesforceMultiObjects-batchsource.md [hunk 3]
```diff
@@ -18,6 +18,10 @@ Configuration
 
 **Password:** Salesforce password.
 
+**Security Token:** Salesforce security Token. If the password does not contain the security token the plugin 
+will append the token before authenticating with salesforce.
+
+
```
Comment (rmstar) [inline docs/SalesforceMultiObjects-batchsource.md]: nit: remove extra newline

File: pom.xml [hunk 4]
```diff
@@ -30,13 +30,13 @@
     <surefire.redirectTestOutputToFile>true</surefire.redirectTestOutputToFile>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <!-- version properties -->
-    <cdap.version>6.2.0-SNAPSHOT</cdap.version>
+    <cdap.version>6.2.3</cdap.version>
```
Comment (rmstar) [inline pom.xml]: Is this plugin only compatible with CDAP 6.2.3 and later versions? If not, I would just remove the -SNAPSHOT suffix. Same thing for hydrator.version.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java [hunk 5]
```diff
@@ -83,12 +83,15 @@ public void prepareRun(BatchSourceContext context) throws ConnectionException {
     String sObjectNameField = config.getSObjectNameField();
     context.setInput(Input.of(config.referenceName, new SalesforceInputFormatProvider(
       config, queries, getSchemaWithNameField(sObjectNameField, schemas), sObjectNameField)));
-
+    /* TODO PLUGIN-510
+     *  As part of [CDAP-16290], recordLineage function was introduced with out implementation.
+     *  To avoid compilation errors the code block is commented for future fix.
     Schema schema = context.getInputSchema();
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java:89]: Why did you need to do this? Your changes look unrelated to lineage

File: pom.xml (hunk 4 above)
Comment (shiva-cirus) [reply pom.xml]: The plugin is compatible with 6.1, 6.2 and 6.3. Let me know what do you want to me to set as default in the pom.xml for both CDAP and Hydrator versions.

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java (hunk 5 above)
Comment (shiva-cirus) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java:89]: Probably you did not follow the google chat on this topic.  To summarize , The current DEVELOP branch does not compile. There was a change introduced last time that caused this. So @CuriousVini  asked to comment the code and add a  TODO.

File: docs/Salesforce-batchsource.md (hunk 1 above)
Comment (shiva-cirus) [reply docs/Salesforce-batchsource.md]: Done

File: docs/Salesforce-streamingsource.md (hunk 2 above)
Comment (shiva-cirus) [reply docs/Salesforce-streamingsource.md]: Done

File: docs/SalesforceMultiObjects-batchsource.md (hunk 3 above)
Comment (shiva-cirus) [reply docs/SalesforceMultiObjects-batchsource.md]: Done

File: pom.xml (hunk 4 above)
Comment (rmstar) [reply pom.xml]: It was 6.2.0-SNAPSHOT before your change, so let's set it to 6.2.0

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java (hunk 5 above)
Comment (rmstar) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBatchMultiSource.java:89]: I see, thanks

File: pom.xml (hunk 4 above)
Comment (shiva-cirus) [reply pom.xml]: Done.

**Comment by shiva-cirus** [comment]
I have implemented all PR comments and also have squashed all the commits to a single commit.

**Review by rmstar — APPROVED** [review]
